### PR TITLE
Update e2e testing documentation

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR addresses the testing documentation update task, following the outline posted in pciE2j-aW-p2.

`README.md`
* add new entries in `README.md` table of contents that link to the new documentation pages.
* remove existing content from page as they are addressed by the new documents.

`setup.md`
* migrate environment setup process to this document and expand on it.
* migrate details on configuration files, and point to internal FG page for the process.
* migrate details about environment variables and branch naming strategy to this document.

`environment_variables.md`
* new document split from the old `config.md`.

`config_values.md`
* new document split from the old `config.md`.

`overview.md`
* a high-level overview that outlines the goals of quality squad and e2e testing.

`running-tests.md`
* refactor existing document.
* covers multiple scenarios of developer running tests.

`style-guide.md`
* refactor of existing document.
* add discussion on tags (parallel, jetpack, etc)

`troubleshooting_debugging.md`
* discussion of possible issue faced by those that pulled the repo fresh.

`writing_tests.md`
* new document outlining some major 'components' to consider when writing tests.
* migrated content from old `gutenberg.md` to become a section on this document.

#### Testing instructions

I have submitted my work in my forked repo of `wp-calypso`.

Please go to https://github.com/worldomonation/wp-calypso/tree/update/e2e-documentation/test/e2e to try it out.

Fixes #

### CODE REVIEW

File: test/e2e/README.md
Comment (cathymcbride) [inline test/e2e/README.md]: Should this have a link top something specific?

**Review by cathymcbride — DISMISSED** [review]
One tiny thing to look at, but overall I think it look great!

File: test/e2e/README.md
Comment (worldomonation) [reply test/e2e/README.md]: Yes - I had not filled out the content of `set up user` yet and so I had left it blank. 

It is not something I've done yet, so my plan is to transplant the appropriate content of the FG page (PCYsg-jvX-p2) here in a subsequent push.

**Comment by worldomonation** [comment]
> One tiny thing to look at, but overall I think it look great!

Excellent. I am in the process of updating other sections as I write this, but the content scope is larger than I expected it to be. I will try to reuse as much of the existing documentation where applicable to timebox this work.

File: test/e2e/docs/setup.md
Comment (bsessions85) [inline test/e2e/docs/setup.md]: I would go into less detail here. maybe just refer internal users to the field guide

File: test/e2e/docs/running-tests.md
Comment (bsessions85) [inline test/e2e/docs/running-tests.md]: I am not sure if we should put this here or not. I guess it doesn't matter, but these won't work for external users

File: test/e2e/docs/setup.md
Comment (worldomonation) [reply test/e2e/docs/setup.md]: Got it - I will move this entire section to the FG and on this page, simply make a note about referring to that page.

File: test/e2e/README.md
Comment (worldomonation) [reply test/e2e/README.md]: I've decided to remove the subsection on `set up user` because it references some internal links, so would be much better suited to a consolidated page on FG.

File: test/e2e/docs/setup.md
Comment (worldomonation) [reply test/e2e/docs/setup.md]: Removed with subsequent pushes.

File: test/e2e/docs/running-tests.md
Comment (worldomonation) [reply test/e2e/docs/running-tests.md]: If it does not have to strictly remain internal, I would prefer to leave here - would prefer to see GitHub documentation be the ongoing, living documentation of e2e tests and this would align with the goal.

With that said, I am open to moving this to a consolidated FG page as well.

**Comment by worldomonation** [comment]
I've completed updating all sections on GitHub.

Some items are not as fleshed out as I would like 
eg. information on canary, writing e2e tests themselves.

But I think this structure will form a solid foundation onto which we can continue expanding on.

File: test/e2e/docs/running-tests.md
Comment (bsessions85) [reply test/e2e/docs/running-tests.md]: I'm fine with leaving it.

**Comment by worldomonation** [comment]
I am ready to merge this in once approved! I feel like I've covered most angles, and if anything else comes up short it can be added later.

**Review by bsessions85 — APPROVED** [review]
This looks really good! We can always adjust where needed later.

**Comment by worldomonation** [comment]
Closing this as I've pushed the branch to this repository itself (instead of a fork). Corresponding PR is #49496.
.

**Pull request closed without merging**
